### DISCUSSION TITLE
Texture file import from VPSC - works in MTEX 5.8.2 but not in MTEX 5.10.0

### DISCUSSION THREAD

**Discussion by Laurah11** [mtex-toolbox/mtex#1908] [Ask Anything] [answered]
Hi everyone,

I was trying out the latest version of MTEX when I discovered version 5.10.0 reads Texture VPSC files differently. I'm not sure how, but a TEX_PHI1.OUT file that I can open with version 5.8.2 is incompatible with the latest version.

I have tried this following lines : 

```matlab
cs_bcc = crystalSymmetry('432', [2.866 2.866 2.866], 'mineral', 'Iron bcc'); %Crystal symetry

pname = 'C:\path'; % path name 

fname = [pname '\TEX_PH1.OUT']; %file name

odf_vpsc = loadODF_VPSC(fname,'halfwidth',10*degree,'CS',cs_bcc); %import
``` 

It works fine with the 5.8.2 version but leads to this errors with the 5.10.0 version : 
![image](https://github.com/mtex-toolbox/mtex/assets/134601571/8ade3f16-7605-4b21-80d6-fafbf814770b)

So far, I haven't found documentation explaining this issue - does anyone have insight on this?

Here is the texture file that I have used if someone want to try : [TEX_PH1.zip](https://github.com/mtex-toolbox/mtex/files/12158963/TEX_PH1.zip)

Thanks in advance,

Laura

**Comment by ralfHielscher** [comment, marked as answer]
Hi Laura,

thank you for reporting this issue. It has been fixed in ca271b731458b10b7aad235b76b13819ed45f408.

All the best,

Ralf.

**Reply by Laurah11** [reply]
Hi Ralf, 

Thank you very much ! With the changement, it works perfecly well. 

All the best, 

Laura

**Reply by ddyyasfh** [reply]
Hi ,

I was trying out the latest version of MTEX when I discovered version 5.10.1 reads Texture VPSC files differently. I'm not sure how, but a TEX_PHI1.OUT file that I can open with version 5.8.0 is incompatible with the latest version.

I have tried this following lines :
![image](https://github.com/mtex-toolbox/mtex/assets/104597723/24e285ae-0ad2-4f66-b7a4-d9cde1e00637)
It works fine with the 5.8.0 version but leads to this errors with the 5.10.1 version :

error vector3d/smooth (line 101)
    clim(sP(j).ax,colorRange);

error vector3d/plot (line 103)
    [varargout{1:nargout}] = v.smooth(varargin{:});

error S2Fun/plot (line 40)
    [ch,cax] = plot(plotNodes,values(:,j),'pcolor','hold',varargin{:});

error S2FunHarmonicSym/plot (lines 26)
[varargout{1:nargout}] = sF.plot@S2Fun(sR,sF.s,varargin{:});

error SO3Fun/plotPDF (line 62)
  [~,cax] = plot(pdf,'smooth','doNotDraw','ensureNonNeg',varargin{:});

error SO3FunHarmonic/plotPDF (line 32)
plotPDF@SO3Fun(SO3F.subSet(1),h,varargin{:});

error texture plot (line 17)
   plotPDF(odf{n},h,'lower','colorbar');

**Reply by ralfHielscher** [reply]
This is not an error with the import. The reason is the command `clim` which was introduced in Matlab 2022a and replaces `caxis`. You can simply replace in  mtex/geometry/@vector3d/smooth (line 101) the command `clim` with `caxis` and everything should be fine.

Ralf.
